extensions/b2g/viewer.js

```diff
@@ -0,0 +1,213 @@
+/* Copyright 2014 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* globals PDFJS, Promise */
+
+'use strict';
+
+PDFJS.useOnlyCssZoom = true;
+PDFJS.disableTextLayer = true;
+PDFJS.workerSrc = '../pdfjs-components/build/pdf.worker.js';
+PDFJS.cMapUrl = '../pdfjs-components/cmaps/';
+PDFJS.cMapPacked = true;
+
+var DEFAULT_SCALE_DELTA = 1.1;
+var MIN_SCALE = 0.25;
+var MAX_SCALE = 10.0;
+
+var PDFViewerApplication = {
+  pdfDocument: null,
+  pdfViewer: null,
+  pdfHistory: null,
+  pdfLinkService: null,
+  loading: true,
+  updateScaleControls: false,
+
+  open: function (params) {
+    var url = params.url, originalUrl = params.originalUrl;
+
+    this.initUI();
+    this.setTitleUsingUrl(originalUrl);
+
+    // Loading document.
+    PDFJS.getDocument(url).then(function (pdfDocument) {
+      // Document loaded, specifying document for the viewer.
+      this.pdfDocument = pdfDocument;
+      this.pdfViewer.setDocument(pdfDocument);
+      this.pdfLinkService.setDocument(pdfDocument);
+      this.pdfHistory.initialize(pdfDocument.fingerprint);
+    }.bind(this));
+  },
+
+  setTitleUsingUrl: function pdfViewSetTitleUsingUrl(url) {
+    this.url = url;
+    var title = PDFJS.getFileName(url) || url;
+    try {
+      title = decodeURIComponent(title);
+    } catch (e) {
+      // decodeURIComponent may throw URIError,
+      // fall back to using the unprocessed url in that case
+    }
+    this.setTitle(title);
+  },
+
+  setTitle: function pdfViewSetTitle(title) {
+    document.title = title;
+    document.getElementById('activityTitle').textContent = title;
+  },
+
+  get pagesCount() {
```

Comment:
Now that we have a link service here and this function (and the other two below this one) are also implemented there, can't we make use of the ones in the link service instead to avoid duplication?